include/eeprom/core/dev_data.hpp

```diff
@@ -286,45 +333,24 @@ class DevDataAccessor
         create_data_part(key, len, dummy);
     }
 
-    void read_complete(uint32_t message_index) final {
-        // we don't need message_index since this is an internal call
-        // and not initatied from a can message
-        std::ignore = message_index;
-        // test if data is set to default
-        auto delivery_state = std::vector<uint8_t>(
-            addresses::lookup_table_tail_length, conf.default_byte_value);
-        if (std::equal(delivery_state.begin(), delivery_state.end(),
-                       data_tail_buff.begin())) {
-            auto init_tail = DataTailType{};
-            std::ignore = bit_utils::int_to_bytes(
-                addresses::data_address_begin, init_tail.begin(),
-                init_tail.begin() + addresses::lookup_table_tail_length);
-            tail_accessor.write(init_tail, 0);
-            data_tail = addresses::data_address_begin;
-        } else {
-            std::ignore = bit_utils::bytes_to_int(
-                data_tail_buff.begin(),
-                data_tail_buff.begin() + addresses::lookup_table_tail_length,
-                data_tail);
-        }
-        tail_updated = true;
-    }
-
     auto data_part_exists(uint16_t key) -> bool {
-        if (tail_updated && config_updated) {
-            auto table_location = calculate_table_entry_start(key);
-            return table_location < data_tail;
+        if (table_ready()) {
+            return calculate_table_entry_start(key) <
+                   tail_accessor.get_data_tail();
         }
         return false;
     }
 
-    auto ready() -> bool { return tail_updated && config_updated; }
+    auto table_ready() -> bool {
+        return config_updated && tail_accessor.get_tail_updated();
```

Comment:
can't this just be `get_tail_updated()` since we're checking whether `config_updated` is true here?

Reply:
Or, we should pull `config_updated` from `get_tail_updated()` and reserve that check for the `table_ready()` function.